multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -42,38 +45,136 @@ func New(cfg *config.Cluster, awsDebug bool) *Cluster {
+			//No results for a list operation is not an error!!! (unless your AWS)
+			stackExists = false
+		} else {
+			return "", fmt.Errorf("error describing stack: %v", err)
+		}
+	} else {
+		stackExists = true
+		if len(describeStacksOutput.Stacks) > 1 {
+			return "", fmt.Errorf("expected error: found more the one stack with name %s", c.ClusterName)
+		}
+	}
+
+	//if the stack already exists, is doesn't make sense to validate the existing VPC for subnet conflicts
+	if c.VPCID != "" && !stackExists {
+		fmt.Println("Existing VPC detected. Will validate for subnet cidr conflicts")
+		if err := c.validateExistingVPC(); err != nil {
+			return "", err
+		}
+	}
+
+	return validationReport.String(), nil
+}
+
+/*
+Calls out to AWS API to get a list of the subnets that already exist in the target existing VPC.
+Ensures that the new instance subnet being created does not overlap with any existing subnets in the VPC.
+*/
+func (c *Cluster) validateExistingVPC() error {
+	ec2Svc := ec2.New(c.session)
+
+	describeVpcsInput := ec2.DescribeVpcsInput{
+		VpcIds: []*string{aws.String(c.VPCID)},
+	}
+	vpcOutput, err := ec2Svc.DescribeVpcs(&describeVpcsInput)
+	if err != nil {
+		return fmt.Errorf("error describing existing vpc: %v", err)
+	}
+	if len(vpcOutput.Vpcs) == 0 {
+		return fmt.Errorf("could not find vpc %s in region %s", c.VPCID, c.Region)
+	}
+	if len(vpcOutput.Vpcs) > 1 {
+		//Theoretically this should never happen. If it does, we probably want to know.
+		return fmt.Errorf("found more than one vpc with id %s. this is NOT NORMAL.", c.VPCID)
+	}
+
+	existingVPC := vpcOutput.Vpcs[0]
+
+	if *existingVPC.CidrBlock != c.VPCCIDR {
+		//If this is the case, our network config validation cannot be trusted and we must abort
+		return fmt.Errorf("configured vpcCidr (%s) does not match actual existing vpc cidr (%s)", c.VPCCIDR, *existingVPC.CidrBlock)
+	}
+
+	describeSubnetsInput := ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{existingVPC.VpcId},
+			},
+		},
+	}
+
+	subnetOutput, err := ec2Svc.DescribeSubnets(&describeSubnetsInput)
+	if err != nil {
+		return fmt.Errorf("error describing subnets for vpc: %v", err)
+	}
+
+	//Config validation has already ensured this subnet is contained by the existing VPC CIDR
+	//We need access to the net objects so we can detect conflicts
+	instanceIP, instanceNet, err := net.ParseCIDR(c.InstanceCIDR)
+	if err != nil {
+		return fmt.Errorf("error parsing instances cidr %s : %v", c.InstanceCIDR, err)
+	}
+
+	//Loop through all existing subnets in the VPC and look for conflicting CIDRS
+	for _, existingSubnet := range subnetOutput.Subnets {
+		existingSubnetIP, existingSubnetNet, err := net.ParseCIDR(*existingSubnet.CidrBlock)
+		if err != nil {
+			return fmt.Errorf("error parsing existing subnet cidr %s : %v", *existingSubnet.CidrBlock, err)
+		}
+
+		if existingSubnetNet.Contains(instanceIP) || instanceNet.Contains(existingSubnetIP) {
```

Comment:
nit: I believe we do this check elsewhere. Might make sense to have a helper function:

```
cidrOverlap(a, b *net.IPNet) bool {
  return a.Contains(b.IP) || b.Contains(a.IP)
}
```

Another nit: `subnetNet` is a strange variable name. existingInstanceNet?